packages/flutter/lib/src/rendering/object.dart

```diff
@@ -819,6 +834,146 @@ class PipelineOwner {
     profile(() { Timeline.finishSync(); });
   }
 
+  /// The maximum number of iterations to use when checking for potential
+  /// incorrect uses of elevation on a [RenderPhysicalModel] or
+  /// [RenderPhysicalShape] with regard to painting order. This number is not
+  /// the same as the number of total [RenderObject]s in the tree, but rather the
+  /// total number of [RenderPhysicalModel] or [RenderPhysicalShape]s to check.
+  ///
+  /// This property is only used when asserts are enabled (e.g. debug mode).
+  ///
+  /// The default value is 10.
+  ///
+  /// See also [debugCheckElevationData].
+  int get maxElevationObjectsToCheck => _maxElevationObjectsToCheck;
+  int _maxElevationObjectsToCheck = 10;
+
+  /// Sets the maximum number of iterations to use when checking for potential
+  /// incorrect uses of elevation on a [RenderPhysicalModel] or
+  /// [RenderPhysicalShape] with regard to painting order. This number is not
+  /// the same as the number of total [RenderObject]s in the tree, but rather the
+  /// total number of [RenderPhysicalModel] or [RenderPhysicalShape]s to check.
+  ///
+  /// Setting this number too high will degrade performance if many render
+  /// objects using elevation are present in the tree all are used correctly.
+  /// See [debugCheckElevationData] for a discussion of the performance
+  /// characterstics of the check.
+  ///
+  /// This property is only used when asserts are enabled (e.g. debug mode).
+  ///
+  /// The default value is 10.
```

Comment:
We should double check this number with stakeholders to see how much elevation stacking they expect.

Reply:
This isn't so much about stacking, but about how many nodes in the render (sub) tree have an elevation property (and are not children of another node that has elevation)